src/setup_payload/QRCodeSetupPayloadGenerator.h

```diff
@@ -82,6 +80,24 @@ class QRCodeSetupPayloadGenerator
      */
     CHIP_ERROR payloadBase38Representation(std::string & base38Representation, uint8_t * tlvDataStart, uint32_t tlvDataStartSize);
 
+    /**
+     * This function is called to encode the binary data of a payload to a
+     * base38 string. The payload's optional data is ignored for compatibility
+     * with devices that don't support std::string or STL.
```

Comment:
How is it OK to ignore optional data? Suggest making it very obvious in name of method that optional TLV data is not included (e.g. `WithoutOptional` or `OmitTlv` or some such suffix).